tests/integration/test_integration_all_rust.py

```diff
@@ -595,229 +590,259 @@ class TestRustWebPush(unittest.TestCase):
+        while not MOCK_SENTRY_QUEUE.empty():
+            MOCK_SENTRY_QUEUE.get_nowait()
+        return
+
     def tearDown(self):
         """Tear down and log processing."""
         process_logs(self)
-        while not MOCK_SENTRY_QUEUE.empty():
-            MOCK_SENTRY_QUEUE.get_nowait()
+        TestRustWebPush.clear_sentry_queue()
 
     def host_endpoint(self, client):
         """Return host endpoint."""
         parsed = urlparse(list(client.channels.values())[0])
-        return "{}://{}".format(parsed.scheme, parsed.netloc)
+        return f"{parsed.scheme}://{parsed.netloc}"
 
-    @inlineCallbacks
-    def quick_register(self):
+    @pytest.mark.asyncio
+    async def quick_register(self):
         """Perform a connection initialization, which includes a new connection,
         `hello`, and channel registration.
         """
-        log.debug("🐍#### Connecting to ws://localhost:{}/".format(CONNECTION_PORT))
-        client = PushTestClient("ws://localhost:{}/".format(CONNECTION_PORT))
-        yield client.connect()
-        yield client.hello()
-        yield client.register()
+        log.debug(f"🐍#### Connecting to ws://localhost:{CONNECTION_PORT}/")
+        client = AsyncPushTestClient(f"ws://localhost:{CONNECTION_PORT}/")
+        await client.connect()
+        await client.hello()
+        await client.register()
         log.debug("🐍 Connected")
-        returnValue(client)
+        return client
 
-    @inlineCallbacks
-    def shut_down(self, client=None):
+    @pytest.mark.asyncio
+    async def shut_down(self, client=None):
         """Shut down client."""
         if client:
-            yield client.disconnect()
+            await client.disconnect()
 
     @property
     def _ws_url(self):
-        return "ws://localhost:{}/".format(CONNECTION_PORT)
+        return f"ws://localhost:{CONNECTION_PORT}/"
 
-    @inlineCallbacks
+    @pytest.mark.asyncio
     @max_logs(conn=4)
-    def test_sentry_output_autoconnect(self):
+    async def test_sentry_output_autoconnect(self):
         """Test sentry output for autoconnect."""
         if os.getenv("SKIP_SENTRY"):
             SkipTest("Skipping sentry test")
             return
         # Ensure bad data doesn't throw errors
-        client = PushTestClient(self._ws_url)
-        yield client.connect()
-        yield client.hello()
-        yield client.send_bad_data()
-        yield self.shut_down(client)
+        TestRustWebPush.clear_sentry_queue()
+        client = AsyncPushTestClient(self._ws_url)
+        await client.connect()
+        await client.hello()
+        await client.send_bad_data()
+        await client.disconnect()
 
         # LogCheck does throw an error every time
-        httpx.get(f"http://localhost:{CONNECTION_PORT}/v1/err/crit", timeout=30)
+        async with httpx.AsyncClient() as httpx_client:
+            await httpx_client.get(f"http://localhost:{CONNECTION_PORT}/v1/err/crit", timeout=30)
```

Comment:
I'd prefer to keep the 30s timeout, if only because CI can be a bit flakey or slow and I'm fine if we don't accidentally timeout and fail the test. If  the `client` responds faster, all the better.